client/rpmtrans.c

```diff
@@ -898,6 +900,20 @@ TDNFTransAddInstallPkg(
         BAIL_ON_TDNF_ERROR(dwError);
     }
 
+    if(pInfo->pszChecksum != NULL) {
+        hash = hash_ops + TDNF_HASH_SHA256;
+
+        dwError = TDNFGetDigestForFile(pszFilePath, hash, digest_from_file);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        if (memcmp(digest_from_file, pInfo->pszChecksum, hash->length))
+        {
+            pr_err("rpm file (%s) checksum (%s) does not match repodata checksum (%s)\n", pszFilePath, digest_from_file, pInfo->pszChecksum);
```

Comment:
You are printing binary data here in the error message. That's why we get failures in the github test run like this:
```
        stdout = out.decode().strip()
>       stderr = err.decode().strip()
E       UnicodeDecodeError: 'utf-8' codec can't decode byte 0xcf in position 137: invalid continuation byte```

But I don't understand why this error message appears in the first place - the checksum should match. And it works when I test this locally.